pkg/asset/manifests/dns.go

```diff
@@ -134,14 +146,37 @@ func (d *DNS) Generate(dependencies asset.Parents) error {
 			}
 		}
 	case gcptypes.Name:
-		if installConfig.Config.Publish == types.ExternalPublishingStrategy {
-			zone, err := icgcp.GetPublicZone(context.TODO(), installConfig.Config.Platform.GCP.ProjectID, installConfig.Config.BaseDomain)
+
+		// Set the public zone
+		switch {
+		case installConfig.Config.Publish != types.ExternalPublishingStrategy:
+			// Do not use a public zone when not publishing externally.
+		case installConfig.Config.GCP.PublicDNSZone.ID != "":
+			// Use the provided zone if specified.
+			project := findGCPProject(installConfig.Config.GCP.PublicDNSZone.ProjectID, installConfig.Config.GCP.ProjectID)
+			config.Spec.PublicZone = &configv1.DNSZone{ID: combineGCPZoneInfo(project, installConfig.Config.GCP.PublicDNSZone.ID)}
```

Comment:
This is currently a breaking change in absence of modifications to the cluster-ingress-operator. It currently only accepts the short name (ID). By forcing all clusters to use the relative name (Name) here, we break the default installs. Instead, I recommend we only use the `combineGCPZoneInfo()` when `installConfig.Config.GCP.PublicDNSZone.ProjectID != "" && installConfig.Config.GCP.PublicDNSZone.ProjectID != installConfig.Config.GCP.ProjectID`

Reply:
Actually, this only applies to the default: use case...